src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/InliningInfoNode.cs

```diff
@@ -68,7 +68,7 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
             {
                 MethodDesc[] inlinees = methodNode.InlinedMethods;
                 MethodDesc inliner = methodNode.Method;
-                EcmaMethod inlinerDefinition = (EcmaMethod)inliner.GetTypicalMethodDefinition();
+                EcmaMethod inlinerDefinition = inliner.GetEcmaDefinition();
```

Comment:
Is this what we want here?

If we inline async thunk but not the method body itself, we do not want to create an artificial dependency on the method body itself.